pkgs/development/go-modules/generic/default.nix

```
@@ -121,12 +117,12 @@ let
   ) // overrideModAttrs modArgs) else "";
 
   package = go.stdenv.mkDerivation (args // {
-    nativeBuildInputs = [ removeReferencesTo go ] ++ nativeBuildInputs;
+    nativeBuildInputs = [ go ] ++ nativeBuildInputs;
 
     inherit (go) GOOS GOARCH;
 
     GO111MODULE = "on";
-    GOFLAGS = "-mod=vendor";
+    GOFLAGS = [ "-mod=vendor" ] ++ lib.optional (!allowGoReference) [ "-trimpath" ];
```

this should be `optionals` since the second argument is a list.
```suggestion
    GOFLAGS = [ "-mod=vendor" ] ++ lib.optionals (!allowGoReference) [ "-trimpath" ];
```

I usually avoid using `optional` because is counterintuitive and often misused https://github.com/NixOS/nixpkgs/pull/60123 https://discourse.nixos.org/t/lib-optional-considered-harmful/7081